dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-fee-record-utilisation.test.ts

```diff
@@ -0,0 +1,198 @@
+import { aTfmFacility, aUtilisationReport } from '../../../../../../test-helpers';
+import { calculateExposure, mapToFeeRecordUtilisation } from './map-to-fee-record-utilisation';
+import * as helpers from '../../../../../helpers';
+
+jest.mock('../../../../../helpers');
+
+describe('map-to-fee-record-utilisation', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('calculateExposure', () => {
+    it('returns the utilisation multiplied by the cover percentage as a decimal', () => {
+      // Arrange
+      const utilisation = 200;
+      const coverPercentage = 80;
+
+      // Act
+      const exposure = calculateExposure(utilisation, coverPercentage);
+
+      // Assert
+      // 200 * 80 / 100 = 160
+      expect(exposure).toEqual(160);
+    });
+
+    it('rounds the exposure to two decimal places', () => {
+      // Arrange
+      const utilisation = 500000.99;
+      const coverPercentage = 85.55;
+
+      // Act
+      const exposure = calculateExposure(utilisation, coverPercentage);
+
+      // Assert
+      // 500000.99 * 85.55 / 100 = 427750.8469... = 427750.85 to 2.d.p
+      expect(exposure).toEqual(427750.85);
+    });
+  });
+
+  describe('mapToFeeRecordUtilisation', () => {
+    it('should set the cover percentage to the corresponding field in the tfm facility', () => {
+      // Arrange
+      const coverPercentage = 80;
+      const facilityId = '12345678';
+
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withFacilityId(facilityId).build();
+
+      const tfmFacility = aTfmFacility();
+      tfmFacility.facilitySnapshot.ukefFacilityId = facilityId;
+      tfmFacility.facilitySnapshot.coverPercentage = coverPercentage;
+
+      // Act
+      const feeRecordUtilisation = mapToFeeRecordUtilisation(feeRecord, tfmFacility);
+
+      // Assert
+      expect(feeRecordUtilisation.coverPercentage).toEqual(coverPercentage);
+    });
+
+    it('should set the value to the corresponding field in the facility snapshot when there are no amendments to the value', () => {
+      // Arrange
+      const value = 200000;
+      const facilityId = '12345678';
+
+      const feeRecord = FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withFacilityId(facilityId).build();
+
+      const tfmFacility = aTfmFacility();
+      tfmFacility.facilitySnapshot.ukefFacilityId = facilityId;
+      tfmFacility.facilitySnapshot.value = value;
+
+      const getLatestCompletedAmendmentFacilityValueSpy = jest.spyOn(helpers, 'getLatestCompletedAmendmentToFacilityValue').mockReturnValue(undefined);
+
+      // Act
+      const feeRecordUtilisation = mapToFeeRecordUtilisation(feeRecord, tfmFacility);
+
+      // Assert
+      expect(getLatestCompletedAmendmentFacilityValueSpy).toHaveBeenCalledTimes(1);
+      expect(getLatestCompletedAmendmentFacilityValueSpy).toHaveBeenCalledWith(tfmFacility);
+      expect(feeRecordUtilisation.value).toEqual(value);
```

Comment:
could this be a separate test? can add the previous bits to a beforeEach
the spy should be one test and the value another

Reply:
Tests with the act outside of them are hard to read and maintain. These are highly related assertions so no point splitting them up.

Reply:
Spoke to fellow civil service devs.  This should be split up.  If something was to fail, it is harder to track it with multiple different assertions in one it block.  the spy should have its own it statement ie the spy has been called and called with values.  And then the return should be a different it block that it returns the expected value @ttbarnes

Reply:
Agree, it's harder to read and maintain when multiple assertions like this are in one "it" block, especially if debugging a failing test; Also means the "it" description blocks align exactly with what it's testing

Reply:
Done